### ISSUE TITLE
S108 - Empty catch blocks are not always bad

### ISSUE THREAD

**Issue by MisinformedDNA** [SonarSource/sonar-csharp#1218]
### Description

I am using an empty catch block and a warning is being added. I disagree that there should be a warning here, empty catch blocks have a lot of uses. I can see maybe warning the user if they are catching `Exception` and ignoring it, but I'm using a more specific Exception type.

### Repro steps

```
try
{
    var uri = new Uri(name);
    if (Regex.IsMatch(uri.Host, "foo"))
        return uri.Host;
}
catch (InvalidOperationException)
{
}

return null;
```

### Expected behavior

No warning is given.

### Actual behavior

Receive S108 warning

### Known workarounds

Add a comment or return statement in the catch block.

### Related information

* SonarC# Version (3.10.0.3095)
* Visual Studio 2017 (15.5.4)

**Comment by fmallet** [comment]
Thanks for your feedback @MisinformedDNA. As you mentioned the rule already supports an exception when the empty block contains a comment. The idea being that if this is really on purpose, the intention should be clarified explicitly in the code by adding a comment. In your example, not sure that what is totally obvious for the author of this piece of code is so quickly obvious for any developer discovering it. Narrowing the scope of this rule to only raise a warning when an `Exception` is caught will lead to generate a good amount of False Negatives. This is always the tricky part when tuning the scope of a rule to find the good trade-off between FN and FP.

**Comment by rami013** [comment]
Hi There,
We have come across the same issue with empty catch blocks in C++. It looks like that exception was only defined for C# rule base in SonarQube, and I was wondering how we can get a similar exception for C++ rule base as well. Could you please help me with that?

**Comment by MisinformedDNA** [comment]
@rami013 You should probably create a new issue.

**Comment by Evangelink** [comment]
+1 on @MisinformedDNA answer. Note that you will need to make it on the right repo, this one is for C# and VB.NET analyzers :)

**Comment by rami013** [comment]
I tried to search for the right repo, but could not find. Any idea which would be the right repo for C/C++ ruleset, please?

**Comment by pavel-mikula-sonarsource** [comment]
https://community.sonarsource.com/ is a place to create such a request.
